src/lib/encoder.ts

```diff
@@ -80,6 +81,55 @@ const decodeDataSourceWithHash = (value: string): URLDataWithHash => {
   return { hashFunction: hashFunction.name, hash: dataHash, url: dataSource };
 };
 
+const encodeCompactBytesArray = (values: string[]): string => {
+  let compactBytesArray = '0x';
+
+  for (let i = 0; i < values.length; i++) {
+    if (values[i] === '')
+      throw new Error(
+        `Couldn't encode, value at index ${i} is '', expected '0x'`,
+      );
+
+    if (!isHex(values[i]))
+      throw new Error(`Couldn't encode, value at index ${i} is not hex`);
+
+    if (values[i].length > 65_535 * 2 + 2)
+      throw new Error(
+        `Couldn't encode bytes[CompactBytesArray], value at index ${i} exceeds 65_535 bytes`,
+      );
+
+    const numberOfBytes = (values[i].length - 2) / 2;
+    const hexNumber = padLeft(numberToHex(numberOfBytes), 4);
+
+    compactBytesArray += stripHexPrefix(hexNumber) + stripHexPrefix(values[i]);
```

Comment:
Here you do `-2` because you want to remove the length of `0x` from the string?
Then you use stripHexPrefix. So you assume all the hex string will start with `0x`. But you are using `isHex` (and not `isHexStrict` https://web3js.readthedocs.io/en/v1.2.11/web3-utils.html#ishexstrict

So I reco:

- Either you use isHexString
- or you make your code work for both 0x and non 0x prefixed hex string.

I recommend the second option.

Reply:
@Hugoo I am also in favour of option 2.
The problem is that in the test suite.

<img width="581" alt="Screenshot 2023-01-24 at 08 44 02" src="https://user-images.githubusercontent.com/31145285/214247475-aa22fbf9-4d3b-4573-8b03-e4a96995c452.png">

it fails because when it decodes, it decodes '' (empty string) as '0x`. And the test states that when decoding it should return the expected values mentioned under the `decodedValue`. So '' (empty string) never match and is always expected to be decoded as `0x`

<img width="564" alt="image" src="https://user-images.githubusercontent.com/31145285/214247208-985e60c1-5307-408f-b119-1b46de5ca515.png">

Reply:
conclusion: we keep `isHex()`